### PR TITLE
gnrc_netif_pktq: schedule immediately if timer us is equal to zero

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR schedules dequeues in `gnrc_netif_pktq` immediately if `CONFIG_GNRC_NETIF_PKTQ_TIMER_US` is equal to zero.
This avoids an extra context switch.

This PR also sets the default timer to zero in order to schedule a packet immediately if the lower layer (either network device or dedicated MAC layer) is busy. This reduces the RTT considerably when using MAC layers that might return -EBUSY if the transmit function is called inside the RX function (e.g SubMAC and hopefully soon [OpenDSME](https://opendsme.org))

### Testing procedure
This was tested using the reworked version of the SubMAC, but it could be tested with current master as well. Make sure pinging large packets (e.g 1024) works as expected using GNRC_NETIF_PKTQ.

### Issues/PRs references
None so far

### CODE REVIEW

File: sys/include/net/gnrc/netif/conf.h [hunk 1]
```diff
@@ -76,7 +76,7 @@ extern "C" {
  * @see         net_gnrc_netif_pktq
  */
 #ifndef CONFIG_GNRC_NETIF_PKTQ_TIMER_US
-#define CONFIG_GNRC_NETIF_PKTQ_TIMER_US       (5000U)
```
Comment (benpicco) [inline sys/include/net/gnrc/netif/conf.h:79]: Do you know for what reason this large number was chosen in the first place?

Comment (jia200x) [reply sys/include/net/gnrc/netif/conf.h:79]: I'm not sure TBH. Maybe @miri64 could give a hint?
This seems to be in the order of a full IEEE 802.15.4 frame (around 4 ms).

**Comment by benpicco** [comment]
Hm shouldn't that timeout only hit if the driver is not providing `NETDEV_EVENT_RX_COMPLETE` / `NETDEV_EVENT_TX_COMPLETE` events? That's where `_send_queued_pkt()` is usually called, `GNRC_NETIF_PKTQ_DEQUEUE_MSG` appears more like a fall-back to me.

Essentially, this will now poll the driver to send the packet instead of waiting for it to send a state change event.

**Comment by jia200x** [comment]
> Hm shouldn't that timeout only hit if the driver is not providing NETDEV_EVENT_RX_COMPLETE / NETDEV_EVENT_TX_COMPLETE events? That's where _send_queued_pkt() is usually called, GNRC_NETIF_PKTQ_DEQUEUE_MSG appears more like a fall-back to me.

I see the point, but as it is now, drivers that provide RX_DONE and TX_DONE still try to schedule the msg. IMO the scheduling should then be done only if the driver cannot report the states back. Otherwise it will just add delay

File: sys/include/net/gnrc/netif/conf.h (hunk 1 above)
Comment (miri64) [reply sys/include/net/gnrc/netif/conf.h:79]: Yepp, that's exactly why the value was chosen, in case the device does not report not being busy in time.

**Comment by benpicco** [comment]
If you drop 5d1070152c0bc8e09411c72a780da770ec910226 I think this could still be useful

**Comment by jia200x** [comment]
done!

**Review by benpicco — APPROVED** [review]

File: sys/include/net/gnrc/netif/conf.h (hunk 1 above)
Comment (benpicco) [reply sys/include/net/gnrc/netif/conf.h]: arg but that clashes with 802.15.4g devices that have a much longer airtime :confused: 

I can of course manually adjust this, but I don't like having to tweak config values based on the radio I'm using - do we even need this if we have TX done/RX done events?

**Pull request merged**